src/backend/distributed/commands/drop_distributed_table.c

```diff
@@ -148,3 +150,21 @@ MasterRemoveDistributedTableMetadataFromWorkers(Oid relationId, char *schemaName
 	char *deleteDistributionCommand = DistributionDeleteCommand(schemaName, tableName);
 	SendCommandToWorkersWithMetadata(deleteDistributionCommand);
 }
+
+
+/*
+ * notify_constraint_dropped simply calls NotifyUtilityHookConstraintDropped
+ * to set ConstraintDropped to true.
+ * This udf is designed to be called from citus_drop_trigger to tell us we
+ * dropped a table constraint.
+ */
+Datum
+notify_constraint_dropped(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	/* TODO: maybe error out if not called via event trigger ? */
```

Comment:
.

Reply:
ping on this

Reply:
(looking for how we can understrand if we are in drop trigger)

Reply:
check for `f (!CALLED_AS_TRIGGER(fcinfo))` in the code?

Reply:
But I guess we may not be able to do it in this function

Reply:
afaik, this `CALLED_AS_TRIGGER` actually checks if function itself is called as trigger.
But with that (7507d32) commit, we shouldn't even need to error out ?